### PR TITLE
Add DCS permissions

### PR DESCRIPTION
**Pull request by JacoBriers**
Related pull-requests:
https://github.com/IMQS/www/pull/2078
https://github.com/IMQS/common/pull/27
https://github.com/IMQS/imqsauth/pull/57
https://github.com/IMQS/otaku/pull/519
https://github.com/IMQS/maps/pull/86

### CODE REVIEW

File: permissions/generators_static_data.rb [hunk 1]
```diff
@@ -153,51 +153,49 @@ def spaced_to_pascal(spaced)
...
-			return null;
+};
+
+/** @suppress {suspiciousCode} */
+export let permissionsArray: Permission[];
+
+// populate RoleArray
+(function () {
+	permissionsArray = [];
+	for (let key in Permissions) {
```
Comment (FritzOnFire) [inline permissions/generators_static_data.rb]: Change the `let` to a const.
Drop the braces for the `for loop` and the `if` statement.

File: permissions/generators_static_data.rb [hunk 2]
```diff
@@ -153,51 +153,49 @@ def spaced_to_pascal(spaced)
...
+		if (Permissions.hasOwnProperty(key)) {
+			permissionsArray.push(Permissions[key]);
 		}
 	}
+}());
+
+export function getPermissionByID(id: string): Permission {
```
Comment (FritzOnFire) [inline permissions/generators_static_data.rb:193]: The content of this method can be changed to:
```typescript
return permissionsArray.find((permission) => return id === permission.id);
```

**Review by FritzOnFire — APPROVED** [review]
Little nit picky... I know you did not write the code but it will help the code conform to the rest of the app. Of course you can ignore it and just merge.

**Review by johnloyimqs — APPROVED** [review]
LGTM

File: permissions/generators_static_data.rb (hunk 2 above)
Comment (JacoBriers) [reply permissions/generators_static_data.rb]: Method returns a single element, not an array, so will leave it as-is

**Pull request merged**